components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java

```diff
@@ -556,30 +491,172 @@ private static void init(String projectName, String configPath, String deploymen
-        for (int i = 0; i < schemasArray.length; i++) {
-            if (schemasArray[i].equalsIgnoreCase("basic")) {
+        for (String s : schemasArray) {
+            if (s.equalsIgnoreCase("basic")) {
                 basic = true;
-            } else if (schemasArray[i].equalsIgnoreCase("oauth2")) {
+            } else if (s.equalsIgnoreCase("oauth2")) {
                 oauth2 = true;
             }
         }
         if (basic && oauth2) {
             basicAuth.setOptional(true);
             basicAuth.setRequired(false);
-        } else if (basic && !oauth2) {
+        } else if (basic) {
             basicAuth.setRequired(true);
             basicAuth.setOptional(false);
-        } else if (!basic && oauth2) {
+        } else if (oauth2) {
             basicAuth.setOptional(false);
             basicAuth.setRequired(false);
         }
         config.setBasicAuth(basicAuth);
     }
-}
 
+    /**
+     * Set endpoints of publisher, admin, registration and token
+     *
+     * @param token token from config file
+     * @return TokenBuilder modified token to be written back to configuration file
+     */
+    private TokenBuilder setEndpoints(Token token) {
+        //new token values for config(to rewrite configuration file)
+        TokenBuilder configTokenValues = new TokenBuilder();
+
+        boolean isEndPointsNeeded; //if endpoint(s) is empty and not defined
+        boolean isBaseURLNeeded; //if endpoint(s) contains {baseURL} or endPointsNeeded
+        boolean isRestVersionNeeded; //if endpoint(s) contains {restVersion}
+
+        String restVersion = token.getRestVersion();
+        publisherEndpoint = token.getPublisherEndpoint();
+        adminEndpoint = token.getAdminEndpoint();
+        registrationEndpoint = token.getRegistrationEndpoint();
+        tokenEndpoint = token.getTokenEndpoint();
+
+        //copy current token config values
+        configTokenValues.setPublisherEndpoint(publisherEndpoint);
+        configTokenValues.setAdminEndpoint(adminEndpoint);
+        configTokenValues.setRegistrationEndpoint(registrationEndpoint);
+        configTokenValues.setTokenEndpoint(tokenEndpoint);
+        configTokenValues.setRestVersion(restVersion);
+        configTokenValues.setBaseURL(token.getBaseURL());
+
+        isEndPointsNeeded = StringUtils.isEmpty(publisherEndpoint) || StringUtils.isEmpty(adminEndpoint) || StringUtils
+                .isEmpty(registrationEndpoint) || StringUtils.isEmpty(tokenEndpoint);
+
+        isBaseURLNeeded =
+                publisherEndpoint.contains("{baseURL}") || adminEndpoint.contains("{baseURL}") || registrationEndpoint
+                        .contains("{baseURL}") || tokenEndpoint.contains("{baseURL}") || isEndPointsNeeded;
+
+        isRestVersionNeeded = publisherEndpoint.contains("{restVersion}") || adminEndpoint.contains("{restVersion}")
+                || registrationEndpoint.contains("{restVersion}") || isEndPointsNeeded;
```

Comment:
use constant for {restVersion}

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/332/commits/7dec453bb0eb0d526d5b95004e51f7b012efb678